store/sdk/src/main/java/org/apache/carbondata/sdk/file/CarbonSchemaReader.java

```diff
@@ -61,14 +65,121 @@ public static Schema readSchemaInSchemaFile(String schemaFilePath) throws IOExce
     return new Schema(schemaList);
   }
 
+  /**
+   * get carbondata/carbonindex file in path
+   *
+   * @param path carbon file path
+   * @return CarbonFile array
+   */
+  private static CarbonFile[] getCarbonFile(String path, final String extension) {
+    String dataFilePath = path;
+    if (!(dataFilePath.contains(extension))) {
+      CarbonFile[] carbonFiles = FileFactory
+          .getCarbonFile(path)
+          .listFiles(new CarbonFileFilter() {
+            @Override
+            public boolean accept(CarbonFile file) {
+              if (file == null) {
+                return false;
+              }
+              return file.getName().endsWith(extension);
+            }
+          });
+      if (carbonFiles == null || carbonFiles.length < 1) {
+        throw new RuntimeException("Carbon file not exists.");
```

Comment:
Why RunTimeException, IO related failures should throw IOException

Reply:
ok, done